### ISSUE TITLE
How do we use mqtts with ssl/tls certificates and connect to mqtt broker

### ISSUE THREAD

**Issue by Devyami-Apcon** [MaximoLiberata/react-native-mqtt.js-example#1]
This work fantastic in node, however the same thing in react native is breaking.

**Comment by MaximoLiberata** [comment]
Yes, the example doesn't implement TLS certificate validation. I think that you only need to add the file's content in the `ca` attribute in the connection instance, Did you try it?

**Comment by Eonfuzz** [comment]
I don't believe it's that simple, after spending an afternoon playing around with it I believe the WS implementation doesn't support ssl/tls certs, as it's a different protocol entirely.

@Devyami-Apcon  did you ever figure out a way of getting ssl/tls working for a react native approach?

**Comment by MaximoLiberata** [comment]
MQTTS is not available to connect by browser environment, but you can connect to broker through WSS which enable SSL/TSL as default.

To implement SSL/TLS over websocket, you need to set up the broker to support it and connect to that specific port which will stablish websocket connectiom through HTTPS.

Also, you can see a related issue about it here 👇🏻
https://stackoverflow.com/questions/78162170/mqtt-over-secure-websockets-using-mqtt-js

**Comment by Eonfuzz** [comment]
Thanks for responding, long story short I was led down an awfully long rabbit hole because of how misdirective the AWS IoT documentation is. I _was_ looking for a way of using their propheted easy and simple IoT event service on a mobile device, but both their supplied SDK's are bloody awful and nonfunctional at best.

For those that come by in the future, after diving down that same rabbit hole AWS IoT is not the correct solution for React Native implementations, as even if you do get it working (And if you do, you will have to roll your own Paho like implementation or pull out parts you want of AmplifyJs) it does not support any of the features that AWS says is featured and working on mobile. Only realtime events and last testament messages.
